feat(thermocycler-refresh): Add KS103J2G thermistor conversions

fsinapi opened the pull request:
### Summary
Added code to support the KS103J2G thermistor used on the thermocycler-refresh. Includes a CSV pulled from the vendor that contains nominal thermistor data.

### Review requests
Right now, _both_ thermistor tables are being included in projects (at least based on my examining through `objdump`), which is a waste of flash memory. Not sure if there's a good way to selectively include the tables without moving this code out of `common/`- if you have any good ideas for this then I'll happily add them into this PR

fsinapi committed:
Added thermistor table for the KS103J2 thermistor used on the thermocycler-refresh.

fsinapi committed:
Added support in thermistor_conversion for using different thermistors

fsinapi committed:
Fixed bug with lambda function for resistance lookup
- Automated formatting

fsinapi committed:
Added test cases for thermistor conversions

sfoster1 reviewed (changes requested):
Code looks good but yeah I'd be worried about having both tables in flash. Honestly the best thing to do might be to take the table generation itself out of top-level `common`. The thermistor stuff is already template-parametrized to accept which thermistor lookup table you want; leave it to the project to generate the table and instantiate it. We can keep the code shared by putting it in some other shared directory with cmake to generate the table and then have the project explicitly add the file rather than having it be part of the static library.

fsinapi committed:
Changed the thermistor_conversion::Conversion class to be templated. Now different projects can specify their own set of thermistor tables to generate, and everything to do with the table lookups is done through a Functor template. Concepts are neat!
- Also updated Cmake files to generated seperate thermistor tables for each project, with the correct subset of tables (NTC for the heater/shaker, KS for the thermocycler). Common generates both for unit test purposes.

fsinapi committed:
Forgot to format code

fsinapi committed:
Fixed an extra semicolon in generated thermistor tables
- Updated -lint targets to capture generated thermistor tables

fsinapi committed:
Updated heater_thread in simulator to use new converter

fsinapi committed:
Delete erroneously committed file

sfoster1 reviewed (approved):
Nice work!
